spec/models/comment_spec.rb

```diff
@@ -1,5 +1,18 @@
 require 'rails_helper'
 
 RSpec.describe Comment, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'associations' do
+    it { should belong_to(:user) }
+    it { should belong_to(:post) }
+  end
+
+  describe '#update_comments_count_on_create' do
+    it 'updates the comments count on create' do
+      user1 = User.create(name: 'John Doe')
+      post1 = Post.create(author: user, title: 'New post')
+      expect do
+        Comment.create(user: user1, post: post1, text: 'New comment')
+      end.to change { post.comments_counter }.by(1)
+    end
```

Comment:
👏 Fantastic effort on your comprehensive testing! 👏 However, it seems there's a specific test case that is not passing. Let's focus on improving it:

- [ ] **Reviewing and Updating Test Case:** The test case in question doesn't pass, and it appears there might be an issue with the reference to the variable `user` in line number 12. Please review your code, and if the variable should indeed be `user`, ensure that it is defined correctly.
![image](https://github.com/mohamedSabry0/BlogsDir/assets/55840999/2d3dde6c-279a-44bf-94e2-2a252f3566f5)

By addressing this issue, you'll enhance the accuracy of your test cases. Keep up the meticulous testing! 🚀🧪